src/Orleans/Facet/FacetAttribute.cs

```diff
@@ -0,0 +1,12 @@
+﻿
+using System;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Marker attribute for facets
+    /// </summary>
+    public abstract class FacetAttribute : Attribute
```

Comment:
We can do this different, of course.  Just a question of trade offs.

Marker attribute requires attributes to be inherited from it, requiring assemblies defining facet attributes to depend on Orleans.  It also may provide for better error detection since a facet attribute which does not have a mapper would indicate a misconfiguration.  Attributes on arguments in grain constructors is probably not a common enough occurrence for performance concerns from looking up a mapper for every property attribute, but the potential cost should be considered.  Considering all attributes also raises the question of multiple mapped attributes, and how to detect and handle that.

All in all, imo, having a marker interface make the system simpler and more strait forward, and costs very little, but I don't have a strong opinion on this.

@sergeybykov, @ReubenBond, ??, thoughts?